scriptmodules/supplementary/wifi.sh

```diff
@@ -108,15 +145,44 @@ function connect_wifi() {
         done
     fi
 
-    create_config_wifi "$type" "$essid" "$key"
+    create_${mgmt_tool}_config_wifi "$type" "$essid" "$key" "$iface"
+    gui_connect_wifi "$iface"
+}
 
-    gui_connect_wifi
+function create_nm_config_wifi() {
+    local type="$1"
+    local essid="$2"
+    local key="$3"
+    local dev="$4"
+    local con="RetroPie-WiFi"
+
+    remove_nm_wifi
+    nmcli connection add type wifi ifname "$dev" ssid "$essid" con-name "$con" autoconnect yes
+    # configure security for the connection
+    case $type in
+        wpa)
+            nmcli connection modify "$con" wifi-sec.key-mgmt wpa-psk
+            nmcli connection modify "$con" wifi-sec.psk-flags 0
+            nmcli connection modify "$con" wifi-sec.psk "$key"
+            ;;
+        wep)
+            nmcli connection modify "$con" wifi-sec.key-mgmt none
+            nmcli connection modify "$con" wifi-sec.wep-key-flags 0
+            nmcli connection modify "$con" wifi-sec.wep-key-type 2
+            nmcli connection modify "$con" wifi-sec.wep-key0 "$key"
+            ;;
+        open)
+            ;;
```

Comment:
minor but could probably shorten this by putting `nmcli connection modify "$con"` in a variable

Reply:
That's a remnant from testing various properties with `nmcli modify` until I found that the `key-flags` are required in order for NM to get the credentials from its config file instead of asking the user.

I've merged the `modify` lines now so it's just called once - it still shows each connection parameter so it's easy to modify/add. Let me know if it looks better.

Reply:
Looks good thanks.